packages/cli/lib/deploy.js

```diff
@@ -18,11 +20,100 @@ const logger = pino(pretty({
   ignore: 'hostname,pid'
 }))
 
+async function askWorkspaceDetails (args) {
+  let workspaceType = args.type
+  /* c8 ignore next 9 */
+  if (!workspaceType) {
+    const answer = await inquirer.prompt({
+      type: 'list',
+      name: 'workspaceType',
+      message: 'Select workspace type:',
+      choices: WORKSPACE_TYPES
+    })
+    workspaceType = answer.workspaceType
+  }
+
+  if (!WORKSPACE_TYPES.includes(workspaceType)) {
+    throw new Error(
+      `Invalid workspace type provided: "${workspaceType}". ` +
+      `Type must be one of: ${WORKSPACE_TYPES.join(', ')}.`
+    )
+  }
+
+  let workspaceId = args['workspace-id']
+  /* c8 ignore next 8 */
+  if (!workspaceId) {
+    const answer = await inquirer.prompt({
+      type: 'input',
+      name: 'workspaceId',
+      message: 'Enter workspace id:'
+    })
+    workspaceId = answer.workspaceId
+  }
+
+  if (!UUID_REGEX.test(workspaceId)) {
+    throw new Error('Invalid workspace id provided. Workspace id must be a valid uuid.')
+  }
+
+  let workspaceKey = args['workspace-key']
+  /* c8 ignore next 9 */
+  if (!workspaceKey) {
+    const answer = await inquirer.prompt({
+      type: 'password',
+      name: 'workspaceKey',
+      message: 'Enter workspace key:',
+      mask: '*'
+    })
+    workspaceKey = answer.workspaceKey
+  }
+
+  return {
+    workspaceType,
+    workspaceId,
+    workspaceKey
+  }
+}
+
+async function readWorkspaceDetails (workspaceKeysPath) {
+  /* c8 ignore next 3 */
+  if (!isAbsolute(workspaceKeysPath)) {
+    workspaceKeysPath = relative(process.cwd(), workspaceKeysPath)
```

Comment:
Is this needed? Since we already know this isn't an absolute path, why do we need to create a relative path?

Reply:
fixed